### PR TITLE
Initial opbeans java application

### PR DESCRIPTION
**Pull request by alvarolobato**
TODO:
- [ ] Readme
- [ ] Set memory db by default
- [ ] Data population

### CODE REVIEW

File: opbeans/src/main/java/co/elastic/apm/opbeans/model/ProductType.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import javax.persistence.Id;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+@Entity(name="product_types")
+@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
+public class ProductType {
+	
+	@Id
+    @GeneratedValue
```
Comment (felixbarny) [inline opbeans/src/main/java/co/elastic/apm/opbeans/model/ProductType.java]: nit: use tab

File: opbeans/src/main/java/co/elastic/apm/opbeans/model/Customer.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+package co.elastic.apm.opbeans.model;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+@Entity(name = "customers")
+@JsonIgnoreProperties({ "hibernateLazyInitializer", "handler" })
```
Comment (felixbarny) [inline opbeans/src/main/java/co/elastic/apm/opbeans/model/Customer.java]: Disabling globally by configuring a `ObjectMapper` like so:
```
mapper.setSerializationInclusion(JsonInclude.Include.NON_EMPTY);
mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
```

File: opbeans/src/main/resources/application.properties [hunk 3]
```diff
@@ -0,0 +1,14 @@
+spring.jpa.database=POSTGRESQL
```
Comment (felixbarny) [inline opbeans/src/main/resources/application.properties]: What about using an in-memory database like H2?

File: opbeans/src/main/java/co/elastic/apm/opbeans/model/ProductType.java (hunk 1 above)
Comment (alvarolobato) [reply opbeans/src/main/java/co/elastic/apm/opbeans/model/ProductType.java]: ouch. I'm using STS and I didn't change the code style, will have a look.

File: opbeans/src/main/resources/application.properties (hunk 3 above)
Comment (alvarolobato) [reply opbeans/src/main/resources/application.properties]: Yes, I need to do that part this is mainly focused for my local testing. I'm changing it to in memory db by default.

File: opbeans/src/main/java/co/elastic/apm/opbeans/model/ProductType.java (hunk 1 above)
Comment (alvarolobato) [reply opbeans/src/main/java/co/elastic/apm/opbeans/model/ProductType.java]: STS defaults to tabs 😞

**Comment by alvarolobato** [comment]
I'll add the H2 database in a follow up PR, needs a way to populate the db etc.

**Pull request merged**